src/NLog.Database/DatabaseParameterInfo.cs

```diff
@@ -203,61 +236,15 @@ internal bool SetDbType(IDbDataParameter dbParameter)
             return true;    // DbType not in use
         }
 
-        private static Type TryParseDbType(string dbTypeString)
+        private static Type TryParseDbType(string dbTypeName)
         {
-            string[] dbTypeNames = dbTypeString?.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
-            dbTypeString = dbTypeNames?.Length > 0 ? dbTypeNames[dbTypeNames.Length - 1] : dbTypeString;
+            // retrieve the type name if a full name is given
+            dbTypeName = dbTypeName?.Substring(dbTypeName.LastIndexOf('.') + 1).Trim();
```

Comment:
Will this crash of the string ends with a `.`?

Reply:
Happy new 2022!

Reply:
> Will this crash when the string ends with a `.`?

Good catch. Will add a test and fix this.

Reply:
Actually, it does not crash, the empty string is returned.  
See [this fiddle](https://dotnetfiddle.net/g5QDwY).
[The documentation](https://docs.microsoft.com/en-us/dotnet/api/system.string.substring?view=net-6.0#System_String_Substring_System_Int32_) agrees too:

> [crash if] `startIndex` is less than zero or greater than the length of this instance.

Reply:
Yes indeed. Thanks for the test cases :)